xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableDBListFilterService.xml

```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="XWiki.LiveTableDBListFilterService" locale="">
+  <web>XWiki</web>
+  <name>LiveTableDBListFilterService</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1496336171000</creationDate>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1497000123000</date>
+  <contentUpdateDate>1496999359000</contentUpdateDate>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
+  <content>{{velocity wiki="false"}}
+#set($reqClass = "$!request.get('class')")
+#set($reqProp = "$!request.get('prop')")
+#set($reqInput = "$!request.get('input')")
+#set($reqLimit = "$!request.get('limit')")
+#if ($reqLimit != '')
+  #set ($limit = $numbertool.integer($reqLimit))
+#else
+  ## The limit is hardcoded here, as we are only dealing with suggestions
+  #set ($limit = 25)
+#end
+#if ($reqClass != '' &amp;&amp; $reqProp != '' &amp;&amp; $reqInput != '')
+  #set ($discard = $response.setContentType('application/json'))
+  #set ($classRef = $xwiki.getClass($reqClass))
+  #set ($field = $classRef.get($reqProp))
+  #if ($field.classType == 'DBList')
+    #set ($sql = $!field.getProperty('sql').value)
+    #set ($results = [])
+    #if ($sql == '')
+      ## Use SOLR
+      #set ($dbId = "$!field.getProperty('idField').value")
+      #set ($dbValue = "$!field.getProperty('valueField').value")
+      #set ($dbClass = $services.model.serialize("$!field.getProperty('classname').value", 'solr'))
+      #set ($solr = "property.${dbClass}.${dbValue}:$reqInput")
```

Comment:
This can be easily broken if the user input has special characters. A better solution is to replace

    q = author:mflorea

with

    q = mflorea
    qf = author

So instead of putting the constraint in the query, you limit the "query fields" (where to search).

Also, the syntax "className.propertyName" is an implementation detail of the [``SolrFieldStringEntityReferenceSerializer``](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/SolrFieldStringEntityReferenceSerializer.java). It doesn't make much sense to pass a String to ``serialize``. You should pass a class property reference (that you create with ``$services.model``) and just append the result to 'property.'